FindFFTW.cmake

```diff
@@ -201,139 +218,228 @@ else()
-  set(FFTW_LIBRARIES ${FFTW_LIBRARIES} ${FFTW_LONGDOUBLE_THREADS_LIB})
-  add_library(FFTW::LongDoubleThreads INTERFACE IMPORTED)
-  set_target_properties(FFTW::LongDoubleThreads
-    PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${FFTW_INCLUDE_DIRS}"
-               INTERFACE_LINK_LIBRARIES "${FFTW_LONGDOUBLE_THREADS_LIB}"
+  find_path(FFTW_INCLUDE_PATH
+    NAMES "fftw3.h"
+    PATHS ${FFTW_INC_SEARCHPATH}
+    NO_DEFAULT_PATH # so it finds the fftw.h file in the MKL directory
   )
-else()
-  set(FFTW_LONGDOUBLE_THREADS_LIB_FOUND FALSE)
-endif()
 
-if (FFTW_DOUBLE_OPENMP_LIB)
-  set(FFTW_DOUBLE_OPENMP_LIB_FOUND TRUE)
-  set(FFTW_LIBRARIES ${FFTW_LIBRARIES} ${FFTW_DOUBLE_OPENMP_LIB})
-  add_library(FFTW::DoubleOpenMP INTERFACE IMPORTED)
-  set_target_properties(FFTW::DoubleOpenMP
-    PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${FFTW_INCLUDE_DIRS}"
-               INTERFACE_LINK_LIBRARIES "${FFTW_DOUBLE_OPENMP_LIB}"
-  )
-else()
-  set(FFTW_DOUBLE_OPENMP_LIB_FOUND FALSE)
-endif()
+  if( FFTW_INCLUDE_PATH )
+    file(TO_CMAKE_PATH "${FFTW_INCLUDE_PATH}" FFTW_INCLUDE_PATH)
+    set(FFTW_INCLUDE_DIRS ${FFTW_INCLUDE_PATH})
+  endif()
+
+  # Create list of FFTW libraries to find
+  if(CMAKE_SIZEOF_VOID_P EQUAL "8")
+    # 64-bit OS
+    if(APPLE)
+      set(FFTW_LIB_SEARCHPATH ${FFTW_ROOT}/lib)
+    else()
+      set(FFTW_LIB_SEARCHPATH ${FFTW_ROOT}/lib/intel64)
+    endif()
+    set(MKL_LIBRARY mkl_intel_ilp64)
+    if(WIN32)
+      set(MKL_OPTIONS /DMKL_ILP64)
+    else()
+      set(MKL_OPTIONS -DMKL_ILP64 -m64)
+    endif()
+  
+  else()
+    # not 64-bit OS
+    set(FFTW_LIB_SEARCHPATH ${FFTW_ROOT}/lib/ia32)
+    if(WIN)
+      set(MKL_LIBRARY mkl_intel_c)
+    else()
+      set(MKL_LIBRARY mkl_intel)
+    endif()
+  endif()
+
+  set(MKL_EXTRA_LIBRARIES mkl_core)
+  list(APPEND MKL_EXTRA_LIBRARIES mkl_sequential)
+  set(FFTW_LIBRARY_NAMES ${MKL_LIBRARY} ${MKL_EXTRA_LIBRARIES})
+  
+  macro(FFTWD_LIB_START)
+    unset(FFTWD_LIB CACHE)
+    unset(FFTWF_LIB CACHE)
+    unset(FFTWD_THREADS_LIB CACHE)
+    unset(FFTWF_THREADS_LIB CACHE)
+    if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+      set(FFTWD_LIB -Wl,--start-group)
+    endif()
+  endmacro()
+
+  macro(FFTWD_LIB_END)
+    if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+      list(APPEND FFTWD_LIB -Wl,--end-group -lpthread -lm -ldl)
+    endif()
+  endmacro()
+
+  # Find each library in FFTW_LIBRARY_NAMES
+  FFTWD_LIB_START()
+  foreach(LIB ${FFTW_LIBRARY_NAMES})
+    string(TOUPPER ${LIB} LIB_UPPER)
+    mark_as_advanced(${LIB_UPPER}_LIB)
+    set(FFTW_LIBRARIES ${FFTW_LIBRARIES} ${LIB_UPPER}_LIB)
+    set(LIB_NAME ${CMAKE_STATIC_LIBRARY_PREFIX}${LIB}${CMAKE_STATIC_LIBRARY_SUFFIX})
+    find_library(${LIB_UPPER}_LIB ${LIB_NAME} ${FFTW_LIB_SEARCHPATH})
+    if(${LIB_UPPER}_LIB)
+      set(${LIB}_FOUND TRUE)
+      list(APPEND FFTWD_LIB ${${LIB_UPPER}_LIB})
+    else()
+      message(FATAL_ERROR "${LIB_NAME} not found.")
+    endif()
+  endforeach()
+  FFTWD_LIB_END()
+  add_compile_options(${MKL_OPTIONS})
```

Comment:
Also here, see comment about compile options. It looks like this may be a different set of options, correct? So maybe put it in a different output variable, e.g. `FFTW_MKL_COMPILE_OPTIONS`.

Reply:
Removed.